bftengine/src/preprocessor/messages/ClientBatchRequestMsg.cpp

```diff
@@ -60,6 +60,36 @@ void ClientBatchRequestMsg::validate(const ReplicasInfo& repInfo) const {
     LOG_ERROR(logger(), "Message sender is invalid" << KVLOG(senderId()));
     throw std::runtime_error(__PRETTY_FUNCTION__);
   }
+  if (!checkElements()) {
+    LOG_ERROR(logger(), "One or more ClientMsg in the list is invalid");
+    throw std::runtime_error(__PRETTY_FUNCTION__);
+  }
+}
+bool ClientBatchRequestMsg::checkElements() const {
+  const auto totalMsgSize = size();
+  const auto& numOfMessagesInBatch = msgBody()->numOfMessagesInBatch;
+  char* dataPosition = body() + sizeof(ClientBatchRequestMsgHeader) + msgBody()->cidSize;
+  auto sigManager = SigManager::instance();
+  bool isClientTransactionSigningEnabled = sigManager->isClientTransactionSigningEnabled();
+  for (auto i = 0u; i < numOfMessagesInBatch; i++) {
+    const auto& singleMsgHeader = *(ClientRequestMsgHeader*)dataPosition;
+    PrincipalId clientId = singleMsgHeader.idOfClientProxy;
+    auto expectedSigLen = (isClientTransactionSigningEnabled ? sigManager->getSigLength(clientId) : 0);
```

Comment:
just a comment: are you aware that tx signing can be enabled, and clientid is not recognized, and the function getSigLength will return 0? The resut will be the same (go out with error) but for different reason - bad client id. Luckily, an error is printed inside the function so you don't need to do anything special here.

Reply:
Thanks for the info